src/solvers/convex_bundle_method.jl

```diff
@@ -0,0 +1,586 @@
+function (b::StopWhenLagrangeMultiplierLess{Nothing,R})(
+    mp::AbstractManoptProblem, bms::ConvexBundleMethodState, i::Int
+) where {R}
+    if i == 0 # reset on init
+        b.reason = ""
+        b.at_iteration = 0
+    end
+    M = get_manifold(mp)
+    if -bms.ξ ≤ b.tol_lag_mult && i > 0
+        b.reason = "After $i iterations the algorithm reached an approximate critical point: the parameter -ξ = $(-bms.ξ) is less than $(b.tol_lag_mult).\n"
+        b.at_iteration = i
+        return true
+    end
+    return false
+end
+function (b::StopWhenLagrangeMultiplierLess{T,Nothing})(
+    mp::AbstractManoptProblem, bms::ConvexBundleMethodState, i::Int
+) where {T}
+    if i == 0 # reset on init
+        b.reason = ""
+        b.at_iteration = 0
+    end
+    M = get_manifold(mp)
+    if (bms.ε ≤ b.tol_errors && norm(M, bms.p_last_serious, bms.g) ≤ b.tol_search_dir) &&
+        i > 0
+        b.reason = "After $i iterations the algorithm reached an approximate critical point: the parameter ε = $(bms.ε) is less than $(b.tol_errors) and |g| = $(norm(M, bms.p_last_serious, bms.g)) is less than $(b.tol_search_dir).\n"
+        b.at_iteration = i
+        return true
+    end
+    return false
+end
+
+function (d::DebugWarnIfLagrangeMultiplierIncreases)(
+    ::AbstractManoptProblem, st::ConvexBundleMethodState, i::Int
+)
+    (i < 1) && (return nothing)
+    if d.status !== :No
+        new_value = -st.ξ
+        if new_value ≥ d.old_value * d.tol
+            @warn """The stopping parameter increased by at least $(d.tol).
+            At iteration #$i the stopping parameter -ξ increased from $(d.old_value) to $(new_value).\n
+            Consider decreasing either the diameter by changing the `diam` keyword argument, or one
+            of the parameters involved in the estimation of the sectional curvature, such as `k_min`,
+            `k_max`, or `ϱ` in the `convex_bundle_method` call.
+            """
+            if d.status === :Once
+                @warn "Further warnings will be supressed, use DebugWarnIfLagrangeMultiplierIncreases(:Always) to get all warnings."
+                d.status = :No
+            end
+        elseif new_value < zero(number_eltype(st.ξ))
+            @warn """The stopping parameter is negative.
+            At iteration #$i the stopping parameter -ξ became negative.\n
+            Consider increasing either the diameter by changing the `diam` keyword argument, or changing
+            one of the parameters involved in the estimation of the sectional curvature, such as `k_min`,
+            `k_max`, or `ϱ` in the `convex_bundle_method` call.
+            """
+        else
+            d.old_value = min(d.old_value, new_value)
+        end
+    end
+    return nothing
+end
+
+function (d::DebugStepsize)(
+    ::P, bms::ConvexBundleMethodState, i::Int
+) where {P<:AbstractManoptProblem}
+    (i < 1) && return nothing
+    Printf.format(d.io, Printf.Format(d.format), get_last_stepsize(bms))
+    return nothing
+end
```

Comment:
I would have thought this should be the default fallback for any state. Is that not the case? Then we should maybe introduce it as a generic default for states.

Reply:
I now see why we need that :)